lib/valkyrie/storage/fedora.rb

```diff
@@ -28,16 +28,16 @@ def find_by(id:)
     # @param resource [Valkyrie::Resource]
     # @return [Valkyrie::StorageAdapter::StreamFile]
     def upload(file:, original_filename:, resource:)
-      # TODO: this is a very naive aproach. Change to PCDM
-      identifier = resource.id.to_uri + '/original'
-      ActiveFedora::File.new(identifier) do |af|
-        af.content = file
-        af.original_name = original_filename
-        af.save!
-        af.metadata.set_value(:type, af.metadata.type + [::RDF::URI('http://pcdm.org/use#OriginalFile')])
-        af.metadata.save
+      identifier = resource_identifier(resource: resource)
+      af = ActiveFedora::File.new(identifier) do |f|
+        f.original_name = original_filename
+        f.content = file
+        f.save!
+        f.metadata.set_value(:type, f.metadata.type + [::RDF::URI('http://pcdm.org/use#OriginalFile')])
+        f.metadata.save
       end
-      find_by(id: Valkyrie::ID.new(identifier.to_s.sub(/^.+\/\//, PROTOCOL)))
+      _version_uri = create_version(af)
+      find_by(id: to_valkyrie_id(identifier))
```

Comment:
I would assume that the `find_by` could cast a given identifier into the appropriate valkyrie id. It seems odd that I have to remember to coerce IDs into their Valkyrie format.